packages/sidecar/src/main.ts

```diff
@@ -50,360 +53,237 @@ type SideCarOptions = {
-      const headers = {
-        ...CORS_HEADERS,
-        ...SPOTLIGHT_HEADERS,
-      };
-      for (const [header, value] of Object.entries(headers)) {
-        res.setHeader(header, value);
-      }
-      if (req.method === "OPTIONS") {
-        res.writeHead(204, {
-          "Cache-Control": "no-cache",
-        });
-        res.end();
-        return;
+const contextId = generateUuidv4();
+const transport = new StreamableHTTPTransport();
+const mcp = createMcpInstance();
+
+export const app = new Hono<HonoEnv>()
+  .use(contextStorage(), async (ctx, next) => {
+    ctx.set("contextId", contextId);
+    await next();
+  })
+  .all(
+    "/mcp",
+    async (_ctx, next) => {
+      if (!mcp.isConnected()) {
+        // Connecting the MCP with the transport
+        await mcp.connect(transport);
       }
-      return handler(req, res, pathname, searchParams);
-    },
-    { name: "enableCORS", op: "sidecar.http.middleware.cors" },
-  );
 
-const streamRequestHandler = (buffer: MessageBuffer<Payload>, incomingPayload?: IncomingPayloadCallback) => {
-  return function handleStreamRequest(
-    req: IncomingMessage,
-    res: ServerResponse,
-    pathname?: string,
-    searchParams?: URLSearchParams,
-  ): void {
-    if (
-      req.method === "GET" &&
-      req.headers.accept &&
-      req.headers.accept === "text/event-stream" &&
-      pathname === "/stream"
-    ) {
-      res.writeHead(200, {
-        "Content-Type": "text/event-stream",
-        "Cache-Control": "no-cache",
-        Connection: "keep-alive",
-      });
-      res.flushHeaders();
-      // Send something in the body to trigger the `open` event
-      // This is mostly for Firefox -- see getsentry/spotlight#376
-      res.write("\n");
-
-      const useBase64 = searchParams?.get("base64") != null;
-      const base64Indicator = useBase64 ? ";base64" : "";
-      const dataWriter = useBase64
-        ? (data: Buffer) => res.write(`data:${data.toString("base64")}\n`)
-        : (data: Buffer) => {
-            // The utf-8 encoding here is wrong and is a hack as we are
-            // sending binary data as utf-8 over SSE which enforces utf-8
-            // encoding. This is only for backwards compatibility
-            for (const line of data.toString("utf-8").split("\n")) {
-              // This is very important - SSE events are delimited by two newlines
-              res.write(`data:${line}\n`);
-            }
-          };
+      await next();
+    },
+    ctx => transport.handleRequest(ctx),
+  )
+  .get("/health", ctx => ctx.text("OK"))
+  .use(cors())
+  .delete("/clear", ctx => {
+    getBuffer().clear();
+    return ctx.text("Cleared");
+  })
+  .get("/stream", ctx => {
+    const buffer = getBuffer();
+
+    const useBase64 = ctx.req.query("base64") != null;
+    const base64Indicator = useBase64 ? ";base64" : "";
+
+    return streamSSE(ctx, async stream => {
       const sub = buffer.subscribe(([payloadType, data]) => {
```

Comment:
I think what would be interesting here, is abstraction sub/unsub, and making this an async generator

possibly a future todo, as i dont know how complicated that is either, but it could make this so it "looks" like a terminating loop that cleans itself up, vs the implicit abstractions that are kind of hard to follow with streamSSE and its child run function